controllers/destination_controller.go

```diff
@@ -92,7 +92,9 @@ func (r *DestinationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return defaultRequeue, nil
 	}
 
-	if err := r.Scheduler.ReconcileDestination(); err != nil {
+	// TODO: Instead of this function call, we could watch for changes to the Destination
+	// CRD and triggering all Works to reconcile on every change.
```

Comment:
question: you mean, inside the scheduler?

if so, that's the end goal (to make the scheduler actually a controller)

Reply:
👌 - the Scheduler watches Destinations and triggers Works to reconcile again on any change (I _think_ that makes sense!)

Reply:
In the future, we want to make the Scheduler a Controller, which watches both Destinations _and_ Works:
- if a new Destination is added, reconcile all Works
- if a new Work is added, reconcile _that_ Work